File: examples/test_simple_class.py
"""Simple class example for v1.2.0"""

class Point:
    """A 2D point class"""
    
    def __init__(self, x: int, y: int):
        self.x = x
        self.y = y
    
    def move_by(self, dx: int, dy: int):
        """Move point by delta"""
        self.x += dx
        self.y += dy
    
    def distance_to(self, other) -> float:
        """Calculate distance to another point"""
        dx = self.x - other.x
        dy = self.y - other.y
        return (dx * dx + dy * dy) ** 0.5

# Test function
def test_point():
    p1 = Point(0, 0)
    p2 = Point(3, 4)
    
    # Move first point
    p1.move_by(1, 1)
    
    # Calculate distance
    dist = p1.distance_to(p2)
    
    return dist

File: examples/test_simple_class.rs
#[derive(Debug, Clone)] pub struct Point {
    pub x: i32, pub y: i32
}
impl Point {
    pub fn new(x: i32, y: i32)  -> Self {
    Self {
    x, y
}
} pub fn move_by(& mut self, dx: i32, dy: i32) {
    self.x = self.x + dx;
    self.y = self.y + dy;
   
}
pub fn distance_to(& self, other: DynamicType)  -> f64 {
    let mut dx = self.x - other.x;
    let mut dy = self.y - other.y;
    return(dx * dx + dy * dy as f64).powf(0.5);
   
}
} #[doc = " Depyler: verified panic-free"] #[doc = " Depyler: proven to terminate"] pub fn test_point()  -> DynamicType {
    let p1 = Point::new(0, 0);
    let p2 = Point::new(3, 4);
    p1.move_by(1, 1);
    let dist = p1.distance_to(p2);
    return dist
}